events/logger/logger.go

```diff
@@ -124,3 +144,88 @@ func processHeaders(hide bool, headers http.Header) http.Header {
 	}
 	return headers
 }
+
+func getSystemInfo() string {
+	builder := strings.Builder{}
+	out, err := os.ReadFile("/etc/os-release")
+	if err == nil {
+		builder.WriteString("OS Info:\n" + string(out) + "\n")
+	}
+	out, err = exec.Command("uname", "-a").CombinedOutput()
+	if err == nil {
+		builder.WriteString("System Info:" + string(out) + "\n")
+	}
+	return builder.String()
+}
+
+// maskIPRouteOutput changes any non-local ip address in the output to ***
+func maskIPRouteOutput(output string) string {
+	expIPv4 := regexp.MustCompile(`\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}`)
+	expIPv6 := regexp.MustCompile(`(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}` +
+		`|([0-9a-fA-F]{1,4}:){1,7}:` +
+		`|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}` +
+		`|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}` +
+		`|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}` +
+		`|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}` +
+		`|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}` +
+		`|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})` +
+		`|:((:[0-9a-fA-F]{1,4}){1,7}|:)` +
+		`|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}` +
+		`|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])` +
+		`|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))`)
```

Comment:
Are you sure it will cover all IPv6 formats? :smile: Maybe we could simplify this somehow? Like - find all fields which contain "." or ":" and mask those? Or cut out specific fields delimited by space? :thinking: , not sure. Just throwing some ideas

Reply:
I did not check this. This is some old untouched code https://github.com/NordSecurity/nordvpn-linux/blob/bb5eb4c0be07ece77b5dc4b255d3afdbcf82302d/daemon/connect.go#L71